File: api-certvalidation/src/tr/gov/tubitak/uekae/esya/api/certificate/validation/find/certificate/trusted/TrustedCertificateFinderFromFileSystem.java
package tr.gov.tubitak.uekae.esya.api.certificate.validation.find.certificate.trusted;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.util.PathUtil;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * Finds trusted certificates from the local file system. 
 */
public class TrustedCertificateFinderFromFileSystem extends TrustedCertificateFinder {
    
    private static final Logger logger = LoggerFactory.getLogger(TrustedCertificateFinderFromFileSystem.class);

    //private static final String DIZIN = "dizin";

    /**
     * Dosyadan Güvenilir Sertifika okur
     */
    protected List<ECertificate> _findTrustedCertificate()
    {
    	List<ECertificate> certList = new ArrayList<ECertificate>();
		
		String dosyaYolu;
		if ((mParameters != null) && ((dosyaYolu = mParameters.getParameterAsString(DOSYA_YOLU)) != null)) 
		{
			try {
				dosyaYolu = PathUtil.getRawPath(dosyaYolu);
				certList.add(ECertificate.readFromFile(dosyaYolu));
			} catch (Exception x) {
				logger.error("Dosyadan (" + dosyaYolu + ")sertifika bilgisi alınamadı", x);
			}
		}

		String dizin;
		if ((mParameters != null) && ((dizin = mParameters.getParameterAsString(DIZIN)) != null)) 
		{
			dizin = PathUtil.getRawPath(dizin);
			File folder = new File(dizin);
			String [] certFileNames = folder.list();
			if(certFileNames != null)
			{
				for (String certFileName : certFileNames) 
				{
					File cert = new File(folder + File.separator + certFileName);
					if(cert.isFile())
					{
						try
						{
							certList.add(new ECertificate(cert));
						}
						catch(Exception e)
						{
							logger.error("Dosyadan sertifika bilgisi alınamadı", e);
						}
					}
				}
			}
		}

		return certList;
    }

}

File: api-certvalidation/csharp/src/find/certificate/trusted/TrustedCertificateFinderFromFileSystem.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Reflection;
using log4net;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.common.util;

namespace tr.gov.tubitak.uekae.esya.api.certificate.validation.find.certificate.trusted
{
    /**
     * Finds trusted certificates from the local file system. 
     */
    public class TrustedCertificateFinderFromFileSystem : TrustedCertificateFinder
    {
        private static readonly ILog logger = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);

        /**
         * Dosyadan Güvenilir Sertifika okur
         */
        protected override List<ECertificate> _findTrustedCertificate()
        {
            List<ECertificate> certList = new List<ECertificate>();

            String dosyaYolu;
            if ((mParameters != null) && ((dosyaYolu = mParameters.getParameterAsString(DOSYA_YOLU)) != null))
            {
                try
                {
                    dosyaYolu = PathUtil.getRawPath(dosyaYolu);
                    certList.Add(ECertificate.readFromFile(dosyaYolu));
                }
                catch (Exception x)
                {
                    logger.Error("Dosyadan (" + dosyaYolu + ") sertifika bilgisi alinamadi", x);
                }
            }

            String dizinYolu;
            if ((mParameters != null) && ((dizinYolu = mParameters.getParameterAsString(DIZIN)) != null))
            {
                dizinYolu = PathUtil.getRawPath(dizinYolu);
                DirectoryInfo folder = null;
                if (dizinYolu.Length == 0)
                {
                    string currentDirectory = Directory.GetCurrentDirectory();
                    folder = new DirectoryInfo(currentDirectory);
                }
                else
                {
                    folder = new DirectoryInfo(dizinYolu);
                }

                FileInfo[] fileNames = null; 
                if(folder.Exists)
                    fileNames = folder.GetFiles();
                else
                {
                    fileNames = new FileInfo[0];
                    logger.Error(folder.FullName + " dizini bulunamadı.");
                }

                foreach (FileInfo fileName in fileNames)
                {
                    try
                    {
                        certList.Add(new ECertificate(fileName));
                    }
                    catch (Exception e)
                    {
                        logger.Error("Dosyadan (" + fileName.FullName + ") sertifika bilgisi alinamadi", e);
                    }
                }
            }

            return certList;
        }
    }
}
